config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,68 @@
+class Rack::Attack
+  ### Prevent Brute-Force Login Attacks ###
+
+  # The most common brute-force login attack is a brute-force password
+  # attack where an attacker simply tries a large number of emails and
+  # passwords to see if any credentials match.
+  #
+  # Another common method of attack is to use a swarm of computers with
+  # different IPs to try brute-forcing a password for a specific account.
+
+  # Throttle POST requests to /login by IP address
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:logins/ip:#{req.ip}"
+  protected_paths = [
+    "/users", # sign up
+    "/session", # sign in
+    "/passwords" # forgot password
+  ]
+  paths_regex = Regexp.union(protected_paths.map { |path| /\A#{Regexp.escape(path)}\z/ })
+
+  throttle('clearance/ip', limit: 5, period: 20.seconds) do |req|
+    req.ip if req.path =~ paths_regex && req.post?
+  end
+
+  # Throttle POST requests to /login by email param
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:logins/email:#{req.email}"
+  #
+  # Note: This creates a problem where a malicious user could intentionally
+  # throttle logins for another user and force their login requests to be
+  # denied, but that's not very common and shouldn't happen to you. (Knock
+  # on wood!)
```

Comment:
👍 Also, the IP address throttle mitigates the risk of an attacker DOS'ing a particular user's login.